plugins/modules/database/saphana/hana_query.py

```diff
@@ -0,0 +1,208 @@
+    - select user_name from users
+    host: "localhost"
+    autocommit: False
+
+- name: Run several queries from file
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    filepath:
+    - /tmp/HANA_CPU_UtilizationPerCore_2.00.020+.txt
+    - /tmp/HANA.txt
+    host: "localhost"
+
+
+'''
+
+import io
+import csv
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def csv_to_json(rawcsv):
+    lines = rawcsv[:rawcsv.rfind('\n')]
+    reader_raw = csv.DictReader(io.StringIO(lines))
+    reader = [dict((k, v.strip()) for k, v in row.items()) for row in reader_raw]
+    tolist = list(reader)
+    return json.dumps(tolist)
+
+
+def main():
+    global module
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            sid=dict(type='str', required=True),
+            instance=dict(type='str', required=False),
+            encrypted=dict(type='bool', required=False, default=False),
+            host=dict(type='str', required=False),
+            user=dict(type='str', required=False, default="SYSTEM"),
+            password=dict(type='str', required=True, no_log=True),
+            database=dict(type='str', required=False),
+            query=dict(type='list', required=False),
+            filepath=dict(type='list', required=False),
+            autocommit=dict(type='bool', required=False, default=True)
+        )
+    )
+    rc, out, err, out_raw = [0, "", "", ""]
+
+    params = module.params
+
+    sid = (params['sid']).upper()
+    instance = params['instance']
+    user = params['user']
+    password = params['password']
+    autocommit = params['autocommit']
+    host = params['host']
+    database = params['database']
+    encrypted = params['encrypted']
+
+    filepath = params['filepath']
+    query = params['query']
+
+    bin_path = f"/usr/sap/{sid}/HDB{instance}/exe/hdbsql"
+
+    if filepath is not None or query is not None:
+        present = True
+    else:
+        present = False
+
+    try:
+        command = [module.get_bin_path(bin_path, required=True)]
+    except Exception:
+        'hdbsql not found {0}. \r\nSTDOUT: {1}\r\n\r\nSTDERR: {2}'.format(
+            rc, out, err)
+
+    if query is not None:
+        if not isinstance(query, (str, list)):
+            module.fail_json(msg="the query option value must be a string or list, passed %s" % type(query))
+
+        for elem in query:
+            if not isinstance(elem, str):
+                module.fail_json(msg="the elements in query list must be strings, passed '%s' %s" % (elem, type(elem)))
+    if filepath is not None:
+        if not isinstance(filepath, (str, list)):
+            module.fail_json(msg="the query option value must be a string or list, passed %s" % type(filepath))
+        for elem in filepath:
+            if not isinstance(elem, str):
+                module.fail_json(msg="the elements in query list must be strings, passed '%s' %s" % (elem, type(elem)))
```

Comment:
Ansible does that for you (the parameters will either be `None`, or a list with strings in them).
```suggestion
```